tree_test.go

```diff
@@ -33,11 +33,17 @@ import (
 	"errors"
 	"fmt"
 	mRand "math/rand"
+	"os"
 	"sort"
 	"testing"
 	"time"
 )
 
+func TestMain(m *testing.M) {
+	_ = GetConfig()
+	os.Exit(m.Run())
+}
```

Comment:
I don't think that is necessary: once the first test is called, it will initialize the `cfg` variable for all subsequent tests. Not sure what you are trying to achieve here?

Reply:
Avoiding adding `GetConfig()` in each test. 

You can't rely on only doing that in the first test of the file, because if you later do `go test ./... -run=TestSomeParticularOne`, that will panic  if isn't the first one in the file. So that means you have to do `GetConfig()` on every test, which can be avoiding on doing this in `TestMain()`.

Makes sense to you?